federation/pkg/kubefed/init/init.go

```diff
@@ -485,6 +507,21 @@ func createAPIServer(clientset *client.Clientset, namespace, name, image, creden
 		return dep, nil
 	}
 
+	if !cloudEnvironment {
```

Comment:
Shouldn't pvc be optional rather than simply disabled for non-cloud?  I'm pretty sure non-cloud deployments will still require persistent storage.